libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,357 @@
+                                          const void * pBuffer, 
+                                          size_t bytesToRead );
+
+/**
+ * @brief The HTTP Client library transport layer interface.
+ */
+typedef struct HTTPTransportInterface
+{
+    HTTPTransportRecv_t recv;
+    HTTPTransportSend_t send;
+    HTTPNetworkContext_t context;
+} HTTPTransportInterface_t;
+
+/**
+ * @brief The HTTP Client library return status.
+ */
+typedef enum HTTPStatus
+{
+    HTTP_SUCCESS = 0,
+    HTTP_INVALID_PARAMETER,
+    HTTP_NETWORK_ERROR,
+    HTTP_NOT_SUPPORTED,
+    HTTP_PARTIAL_RESPONSE,
+    HTTP_INSUFFICIENT_MEMORY,
+    HTTP_INTERNAL_ERROR,
+    HTTP_SECURITY_ALERT_TOO_MANY_HEADERS,
+    HTTP_SECURITY_ALERT_PARSER_INVALID_CHARACTER,
+    HTTP_SECURITY_ALERT_INVALID_CONTENT_LENGTH,
+    /* TODO: Add return codes as implementation continues. */
+} HTTPStatus_t;
+
+/**
+ * @brief Represents header data that will be sent in an HTTP request. 
+ * 
+ * The memory for the header data buffer is supplied by the user. Information in
+ * the buffer will be filled by calling #HTTPClient_InitializeHeaders.
+ */
+typedef struct HTTPRequestHeaders
+{
+    
+    /**
+     * @brief Buffer to hold the raw HTTP request headers.
+     * 
+     * This buffer is supplied by the application.
+     * 
+     * This buffer is owned by the library during #HTTPClient_AddHeader, 
+     * #HTTPClient_AddRangeHeader, #HTTPClient_InitializeRequestHeaders, and
+     * #HTTPClient_Send. This buffer should not be modifed until 
+     * after these functions return.
+     * 
+     * For optimization this buffer may be re-used with the response. The user 
+     * can re-use this buffer for the storing the response from the server in 
+     * #HTTPResponse_t.pBuffer.
+     */
+    uint8_t* pBuffer;
+    size_t bufferLen; /**< The length of pBuffer in bytes. */
+    
+    /**
+     * @brief The actual amount of headers in the buffer, in bytes. This field
+     * is updated by the HTTP Client library functions #HTTPClient_AddHeader,
+     * and #HTTPClient_InitializeRequestHeaders.
+     */
+    size_t headersLen;
+} HTTPRequestHeaders_t;
+
+/**
+ * @brief Configurations of the initial request headers.
+ */
+typedef struct HTTPRequestInfo
+{
+    /**
+     * @brief The HTTP request method i.e. "GET", "POST", "PUT", or "HEAD". 
+     */
+    const char* method;
+    size_t methodLen; /**< The length of the method in bytes. */
+    
+    /**
+     * @brief The Request-URI to the objects of interest, i.e. "/path/to/item.txt".
+     */
+    const char* pPath;
+    size_t pathLen; /**< The length of the path in bytes. */
+
+    /**
+     * @brief The DNS resolvable name of the host or server, i.e. 
+     * "s3.amazonaws.com"
+     * 
+     * The host does not have a "https://" or "http://" prepending.
+     */
+    const char* pHost;
+    size_t hostLen; /**< The length of the host in bytes. */
+   
+    /**
+     * @brief Flags to activate other request header configurations. 
```

Comment:
What values can this have?  Are the `#define` values for these flags available yet?

Reply:
They are near the top of the file at "Flags for #HTTPResponse_t.flags."